### ISSUE TITLE
Ignore "Not implemented" string for the `StringLiteralDuplication` rule

### ISSUE THREAD

**Issue by igorwojda** [detekt/detekt#5327]
## Expected Behavior of the rule
The below code violates the `StringLiteralDuplication` rule (ok)

```
override fun update() {
        TODO("Not implemented")
    }

    override fun getData() {
        TODO("Not implemented")
    }
```

However `TODO("Not implemented")` is the default implementation when generating interface members using Kotlin plugin.

## Context

In general these todos are not desired in codebase, but TODO usage is guarded by `NotImplementedDeclaration` rule. I believe we should exempt "Not Implemented" string from string duplication check (maybe it only should be exempt for this specific `TODO`  method).

**Comment by cortinico** [comment]
You can use the `ignoreStringsRegex` to achieve this:
https://github.com/detekt/detekt/blob/7eab480468da5eeea3fcf61bf57ce7c7e9350155/detekt-rules-complexity/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/StringLiteralDuplication.kt#L65-L66

**Comment by igorwojda** [comment]
I struggling a bit on how exactly those should be done. This property and config seem to be accepting a signal value, not a list and we already have one value for `$`.  Can you please drop the full config line for the config file?

**Comment by BraisGabin** [comment]
```yaml
ignoreStringsRegex: 'Not implemented'
```

or

```yaml
    ignoreStringsRegex: '^Not implemented$'
```
